### PR TITLE
MAINT: dir(numpy) returns duplicate "testing"

### LINKED ISSUE #15383: dir(numpy) returns duplicate "testing"
### Reproducing code example:

```python
import numpy as np
>>> np.__version__
'1.18.1'
>>> len(dir(np))
620
>>> np.testing
<module 'numpy.testing' from 'C:\\Python\\Python38\\lib\\site-packages\\numpy\\testing\\__init__.py'>
>>> len(dir(np))
621
>>> [i for i in dir(np) if i == "testing"]
['testing', 'testing']
```
### Error:
"testing" appears twice in dir(np)

### Numpy/Python version information:

Python 3.8.0 (tags/v3.8.0:fa919fd, Oct 14 2019, 19:37:50) [MSC v.1916 64 bit (AMD64)] on win32
>>> np.__version__
'1.18.1'

### PR DESCRIPTION
**Pull request by SanthoshBala18**
Fixes #15383 

Modified `__dir__()` to remove duplicate "Tester/Testing" attribute. Also added a test to verify this.

### CODE REVIEW

File: numpy/tests/test_public_api.py [hunk 1]
```diff
@@ -94,6 +94,19 @@ def test_import_lazy_import(name):
     assert name in dir(np)
 
 
+@pytest.mark.parametrize('name', ['testing', 'Tester'])
+def test_dir_testing(name):
+    """
+    Test "dir" is not returning duplicate attribute for testing/Tester
+    """
+    exe = (sys.executable, '-c', "import numpy; numpy." + name)
```
Comment (eric-wieser) [inline numpy/tests/test_public_api.py]: What is this for?

Comment (SanthoshBala18) [reply numpy/tests/test_public_api.py]: This is to verify that explicitly calling numpy.testing is not adding a second copy of 'testing' to the result of dir.

Comment (eric-wieser) [reply numpy/tests/test_public_api.py]: But this is started in a separate process, so can have no effect on the rest of the test.

Comment (seberg) [reply numpy/tests/test_public_api.py]: At this point in the test we probably already did anything bad (e.g. testing is definitely imported), wouldn't it be enough to just test that `len(dir(np)) == len(set(dir(np)))`, with a comment of why?

Comment (SanthoshBala18) [reply numpy/tests/test_public_api.py]: @eric-wieser @seberg I have modified the test to assert len(dir(np)) == len(set(dir(np)))

Comment (seberg) [reply numpy/tests/test_public_api.py]: Thanks, seems fair enough to me. If you are still enthusiastic: We try to keep the lines below 80 characters long (so the comment is a bit too long). Might as well rebase/squash the commits (but we can also do both of that).

Comment (SanthoshBala18) [reply numpy/tests/test_public_api.py]: @seberg  Thanks for the review. I have modified the comments to be inline with the standards.

**Comment by seberg** [comment]
Thanks, lets put this in.

**Pull request merged**